### ISSUE TITLE
Stuck at same symbol...

### ISSUE THREAD

**Issue by FireFoxII** [ccxt/binance-trade-bot#407]
When I start the bot, I see first 2/3 days that it switch from symbol to symbol for the best ratio I think...

After about 2/3 days, it stuck at same symbol and I need to restart the bot (delete db file etc...)

**Comment by MasaiasuOse** [comment]
It isn't necessarily stuck. The bot jumps to the worst coin and sometimes the coin can be for a week the worst one ;)

**Comment by Adik128** [comment]
> It isn't necessarily stuck. The bot jumps to the worst coin and sometimes the coin can be for a week the worst one ;)

is it possible to change this algorithm? a week without trade is bad

**Comment by kooltux** [comment]
Same here: the default strategy algorithm pushed some orders but ATM it's "stuck" on OMG... And no more trade for the last 24 hours. It seems surprising, no?

**Comment by donmega** [comment]
Coins can be stuck for weeks. It's part of the strategy. Please read the readme.

**Comment by kooltux** [comment]
The strategy is not that bad, but it's my first time running the bot and I expected more frequent changes (let's say a 1 to 10 transactions per day). And that's a bit surprising because all cryptos seem to move very fast. 

At the end, how can we know what's going on? Would it be possible to add a "scout helper" (=some python code) that could be run in parallel of the main algorithm (using the database as readonly) to extract the significant values that would trigger an order? It's a bit different from backtest because I see that like an observer of "what's going on". Indeed, the database is here so anyone can run SQL queries on it, but it's more difficult.

The current log file for the bot is minimal and perfectly fine, but while "scouting", the operator (=me!) is a bit lost. Except by debugging python, I have no way to understand what's going on.

For sure, what I said is not related to any issue, but it's rather an improvement proposal.

And BTW, great job. Always nice to hear I hope :)

**Comment by donmega** [comment]
The goal of the bot is not to gain more fiat, but more coins. It can stay at the same coin for days and sometimes weeks. It's just the way the strategy works.

Remember: It's not about fiat with this bot. It's about gaining coins. The bot doesn't even understand fiat.

**Comment by kooltux** [comment]
Yes, gaining coins is definitely the goal, so using stable coins as a bridge makes sense, definitely. And also, you may know that in France, as soon as you convert coins to fiat, you're subject to taxes => at least for us, we always stay in crypto world for trading and only pay taxes when you cash-out.
My point in the previous message is not on anything wrong with the default strategy. It's just a bit surprising. So having a kind of basic monitoring would help. If it boils down to run some sqlite queries regularly to understand what's going on, it's fine! But ATM, you can only see "I am scouting the best trades. Current coin:XXXXX" and that's all. Don't you think that some kind of introspection in the trading algorithm would be interesting?

**Comment by MasaiasuOse** [comment]
Scroll the read.me to the bottom. You can find a section named related projects. There is this repo: https://github.com/lorcalhost/BTB-manager-telegram

You can use this to gain additional information, change config, and start/stop the bot

**Comment by kooltux** [comment]
Thx @MasaiasuOse ! But the read.me doesn't help: I took a look but using Telegram is a show stopper. Sorry. I have no trust in Telegram and it's not fully OSS. On my side, I use Matrix, which is a bit better from OSS and security POV (at least you can have your own server!).

At the end, I don't understand why Telegram and other messaging tools like Slack and others are considered as alpha and omega in this trading ecosystem: I don't trust Telegram, Slack or any other network and I need something which doesn't depend on anyone. Gotify is a better approach for these kind of private notifications, even if you need to own your server (which is easy). And remember that relying on somebody else to act is a bad solution. Some stupid people would pay some fees for that, and some others think it's free... Don't trust anyone!

That said, what I meant is a monitoring tool to help to understand/develop various strategies, not just following orders (which you'll get in binance logs anyway...)

So, if we can think about a simple CLI program to monitor operations on your own bot, I'm fine with it. For notifications, using apprise is ok as anyone should be able to configure where the notifications are pushed (be it on Telegram or not!)
